clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -848,6 +848,8 @@ public void handleResponse(AbstractResponse response) {
                         coordinatorReloaded = true;
                         lookupCoordinator(FindCoordinatorRequest.CoordinatorType.GROUP, builder.consumerGroupId());
                     }
+                } else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
```

Comment:
Never mind. The issue here is that the broker potentially has stale metadata, not us, so refreshing wouldn't help.